js/views/conversation_view.js

```diff
@@ -523,18 +534,16 @@
             var collection = this.model.messageCollection;
             var length = collection.length;
             var viewportBottom = this.view.outerHeight;
-            var unreadCount = this.model.get('unreadCount');
-
-            if (!unreadCount || unreadCount < 1) {
-                return;
-            }
+            var unreadCount = this.model.get('unreadCount') || 0;
 
             // Start with the most recent message, search backwards in time
             var foundUnread = 0;
             for (var i = length - 1; i >= 0; i -= 1) {
-                // We don't want to search through all messages, so we stop after we've
-                //   hit all unread messages. The unread should be relatively recent.
-                if (foundUnread >= unreadCount) {
+                // Search the latest 30, then stop if we believe we've covered all known
+                //   unread messages. The unread should be relatively recent.
+                // Why? local notifications can be unread but won't be reflected the
+                //   conversation's unread count.
+                if (i > 30 && foundUnread >= unreadCount) {
```

Comment:
I can convince myself that this will work most of the time, but 30 does feel a little arbitrary.

Not blocking, but some questions to think about in the longer term:
1. How bad is the performance hit if we iterate over all messages
2. Can we refactor, leverage a database index, and/or address performance of this operation in some other way.